src/Native/Time.js

```diff
@@ -10,39 +10,42 @@ Elm.Native.Time.make = function(elm) {
   var Maybe = Elm.Maybe.make(elm);
   var Utils = Elm.Native.Utils.make(elm);
 
+  function fst(pair) {
+      return pair._0;
+  }
+
+  function snd(pair) {
+      return pair._1;
+  }
+
   function fpsWhen(desiredFPS, isOn) {
     var msPerFrame = 1000 / desiredFPS;
-    var prev = elm.timer.now(), curr = prev, diff = 0, wasOn = true;
-    var ticker = NS.input(diff);
-    function tick(zero) {
-      return function() {
-        curr = elm.timer.now();
-        diff = zero ? 0 : curr - prev;
-        if (prev > curr) {
-          diff = 0;
-        }
-        prev = curr;
-        elm.notify(ticker.id, diff);
-      };
-    }
+    var wasOn = true;
+    var ticker = NS.input(true);
     var timeoutID = 0;
     function f(isOn, t) {
       if (isOn) {
-        timeoutID = elm.setTimeout(tick(!wasOn && isOn), msPerFrame);
+        timeoutID = elm.setTimeout( function() { elm.notify(ticker.id, !wasOn && isOn); } , msPerFrame);
       } else if (wasOn) {
         clearTimeout(timeoutID);
       }
       wasOn = isOn;
       return t;
     }
-    return A3( Signal.map2, F2(f), isOn, ticker );
+    function g(event, old) {
+      var curr = fst(event);
+      return Utils.Tuple2(snd(event) ? 0 : curr - snd(old), curr);
+    }
+    var deltas = A2( Signal.map, fst, A3( Signal.foldp, F2(g), Utils.Tuple2(0, elm.timer.programStart), NS.timestamp(ticker) ) );
+    return A3( Signal.map2, F2(f), isOn, deltas );
   }
```

Comment:
These last few lines are really hard to read for me.
 * `f` and `g` should have proper names. (I thought of `startStopTimer` and `calcDelta`, but you can choose).
 * The tuple coming out of `NS.timestamp` and the tuple from `g` are not the same, but I thought they were. Since this will not be exposed to Elm, you can use an object, it doesn't have to be an Elm type: `{ delta: 0, timestamp: elm.timer.programStart }`